src/huggingface_hub/inference/_providers/fal_ai.py

```diff
@@ -82,9 +91,41 @@ class FalAITextToVideoTask(FalAITask):
     def __init__(self):
         super().__init__("text-to-video")
 
+    def _prepare_route(self, mapped_model: str) -> str:
+        if self.api_key.startswith("hf_"):
+            # Use the queue subdomain for HF routing
+            return f"/{mapped_model}?_subdomain=queue"
+        return f"/{mapped_model}"
+
     def _prepare_payload_as_dict(self, inputs: Any, parameters: Dict, mapped_model: str) -> Optional[Dict]:
         return {"prompt": inputs, **filter_none(parameters)}
 
     def get_response(self, response: Union[bytes, Dict]) -> Any:
+        response_dict = _as_dict(response)
+        session = get_session()
```

Comment:
FYI `get_session()` can be used everywhere without having to store it in a variable since it's a singleton (unique per thread). This is the case in any part of `huggingface_hub` code.

(fine to keep it here, just wanted to highlight it)